client/components/Dashboard.tsx

```diff
@@ -1,129 +1,81 @@
-import React, { useState } from 'react';
-import {
-    Chart as ChartJS,
-    LinearScale,
-    CategoryScale,
-    BarElement,
-    PointElement,
-    LineElement,
-    Legend,
-    Tooltip,
-} from 'chart.js';
-import { Chart } from 'react-chartjs-2';
-import { faker } from '@faker-js/faker';
+import React, { useState, useEffect } from 'react';
+import { useQuery, gql } from '@apollo/client';
 import Logger from './Logger';
+import ChartBox from './ChartBox';
+import { Projects, SelectedProject } from './Interfaces';
 
-ChartJS.register(
-    LinearScale,
-    CategoryScale,
-    BarElement,
-    PointElement,
-    LineElement,
-    Legend,
-    Tooltip
-);
+const GET_USER_DATA = gql`
+    query getUserData($userId: String!) {
+        user(id: $userId) {
+            email
+            password
+            projects {
+                id
+                userID
+                name
+                queries {
+                    id
+                    projectID
+                    name
+                    complexity
+                    depth
+                    time
+                }
+            }
+        }
+    }
+`;
 
-const labels = ['Query1', 'Query2', 'Query3', 'Query4', 'Query5', 'Query6', 'Query7'];
+const GET_PROJECT = gql`
+    query Query($projectId: String!) {
+        project(id: $projectId) {
+            id
+            queries {
+                time
+                depth
+                complexity
+            }
+        }
+    }
+`;
 
-export const data = {
-    labels,
-    datasets: [
-        {
-            type: 'line' as const,
-            label: 'Dataset 1',
-            borderColor: 'rgb(255, 99, 132)',
-            borderWidth: 2,
-            fill: false,
-            data: labels.map(() => faker.datatype.number({ min: -1000, max: 1000 })),
-        },
-        {
-            type: 'bar' as const,
-            label: 'Dataset 2',
-            backgroundColor: 'rgb(75, 192, 192)',
-            data: labels.map(() => faker.datatype.number({ min: -1000, max: 1000 })),
-            borderColor: 'rgb(75, 192, 192)',
-            borderWidth: 2,
+function Dashboard() {
+    const { data: dataR, loading: loadingR } = useQuery(GET_USER_DATA, {
+        variables: {
+            userId: '6286978e12716d47e6884194',
```

Comment:
Are `userId` on line 45 and `projectId` on line 50 hardcoded for a reason? Can they be passed in as props or obtained from a route url?